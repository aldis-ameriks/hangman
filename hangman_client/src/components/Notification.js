import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import amber from '@material-ui/core/colors/amber';
import green from '@material-ui/core/colors/green';
import red from '@material-ui/core/colors/red';
import blue from '@material-ui/core/colors/blue';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import WarningIcon from '@material-ui/icons/Warning';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';

const variants = {
  error: { color: red[600], icon: ErrorIcon },
  info: { color: blue[600], icon: InfoIcon },
  success: { color: green[600], icon: CheckCircleIcon },
  warning: { color: amber[600], icon: WarningIcon }
};

const NotificationStyled = styled.div`
  background-color: ${props => props.color};
  font-size: 1rem;
  padding: 1rem;
  color: white;
  width: 100%;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14),
    0px 1px 18px 0px rgba(0, 0, 0, 0.12);
  border-radius: 3px;
  vertical-align: middle;
  display: flex;
  align-items: center;
`;

const IconStyled = styled.div`
  font-size: 20px;
  vertical-align: middle;
  margin-right: 1rem;
`;

const Notification = ({ children, variant }) => {
  const color = variants[variant].color;
  const Icon = variants[variant].icon;
  return (
    <NotificationStyled color={color}>
      <IconStyled as={Icon} data-testid="icon" />
      {children}
    </NotificationStyled>
  );
};

Notification.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.oneOf(['error', 'success', 'info', 'warning']).isRequired
};

export default Notification;
