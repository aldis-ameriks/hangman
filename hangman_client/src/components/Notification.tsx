import React from 'react';
import styled from 'styled-components';

import amber from '@material-ui/core/colors/amber';
import blue from '@material-ui/core/colors/blue';
import green from '@material-ui/core/colors/green';
import red from '@material-ui/core/colors/red';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';
import WarningIcon from '@material-ui/icons/Warning';

const variants = {
  error: { color: red[600], icon: ErrorIcon },
  info: { color: blue[600], icon: InfoIcon },
  success: { color: green[600], icon: CheckCircleIcon },
  warning: { color: amber[600], icon: WarningIcon },
};

const NotificationStyled = styled.div`
  background-color: ${props => props.color};
  font-size: 1rem;
  padding: 1rem;
  color: white;
  width: 100%;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14),
    0px 1px 18px 0px rgba(0, 0, 0, 0.12);
  border-radius: 3px;
  vertical-align: middle;
  display: flex;
  align-items: center;
`;

const IconStyled = styled.div`
  font-size: 20px;
  vertical-align: middle;
  margin-right: 1rem;
`;

export type NotificationType = 'success' | 'error' | 'warning' | 'info';
type Props = {
  children: React.ReactNode;
  variant: NotificationType;
};

const Notification: React.FunctionComponent<Props> = ({ children, variant }) => {
  const color = variants[variant].color;
  const Icon = variants[variant].icon;
  return (
    <NotificationStyled color={color}>
      <IconStyled as={Icon} data-testid="icon" />
      {children}
    </NotificationStyled>
  );
};

export default Notification;
