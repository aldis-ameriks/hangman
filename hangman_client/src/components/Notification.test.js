import React from 'react';
import { render } from 'react-testing-library';
import Notification from './Notification';

describe('Notification', () => {
  ['error', 'success', 'warning', 'info'].forEach(variant => {
    it(`${variant} is rendered correctly`, () => {
      const component = render(
        <Notification variant={variant}>
          <div>Child</div>
        </Notification>
      );
      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
