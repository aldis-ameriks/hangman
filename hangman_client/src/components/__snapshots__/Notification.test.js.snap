// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification error is rendered correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa bHEuh"
      color="#e53935"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1 sc-bwzfXH QPxvA"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
      <div>
        Child
      </div>
    </div>
  </div>
</body>
`;

exports[`Notification info is rendered correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa jxCSlW"
      color="#1e88e5"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-28 sc-bwzfXH QPxvA"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        />
      </svg>
      <div>
        Child
      </div>
    </div>
  </div>
</body>
`;

exports[`Notification success is rendered correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa ffnKSt"
      color="#43a047"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-10 sc-bwzfXH QPxvA"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <div>
        Child
      </div>
    </div>
  </div>
</body>
`;

exports[`Notification warning is rendered correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa hrQCMv"
      color="#ffb300"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-19 sc-bwzfXH QPxvA"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
      <div>
        Child
      </div>
    </div>
  </div>
</body>
`;
